packages/plugin-axe/src/lib/runner/transform.ts

```diff
@@ -0,0 +1,110 @@
+import type { AxeResults, ImpactValue, NodeResult, Result } from 'axe-core';
+import type axe from 'axe-core';
+import type {
+  AuditOutput,
+  AuditOutputs,
+  Issue,
+  IssueSeverity,
+} from '@code-pushup/models';
+import {
+  countOccurrences,
+  objectToEntries,
+  pluralizeToken,
+  truncateIssueMessage,
+} from '@code-pushup/utils';
+
+/**
+ * Transforms Axe results into audit outputs.
+ * Priority: violations > incomplete > passes > inapplicable
+ */
+export function toAuditOutputs(
+  { passes, violations, incomplete, inapplicable }: AxeResults,
+  url: string,
+): AuditOutputs {
+  const auditMap = new Map<string, AuditOutput>([
+    ...inapplicable.map(res => [res.id, toAuditOutput(res, url, 1)] as const),
+    ...passes.map(res => [res.id, toAuditOutput(res, url, 1)] as const),
+    ...incomplete.map(res => [res.id, toAuditOutput(res, url, 0)] as const),
+    ...violations.map(res => [res.id, toAuditOutput(res, url, 0)] as const),
+  ]);
+
+  return Array.from(auditMap.values());
+}
+
+/**
+ * For failing audits (score 0), includes detailed issues with locations and severities.
+ * For passing audits (score 1), only includes element count.
+ */
+function toAuditOutput(
+  result: Result,
+  url: string,
+  score: number,
+): AuditOutput {
+  const base = {
+    slug: result.id,
+    score,
+    value: result.nodes.length,
+  };
+
+  if (score === 0 && result.nodes.length > 0) {
+    const issues = result.nodes.map(node => toIssue(node, result, url));
+
+    return {
+      ...base,
+      displayValue: formatSeverityCounts(issues),
+      details: { issues },
+    };
+  }
+
+  return {
+    ...base,
+    displayValue: pluralizeToken('element', result.nodes.length),
+  };
+}
+
+function formatSeverityCounts(issues: Issue[]): string {
+  const severityCounts = countOccurrences(
+    issues.map(({ severity }) => severity),
+  );
+
+  return objectToEntries(severityCounts)
+    .toSorted(([a], [b]) => {
+      const order = { error: 0, warning: 1, info: 2 };
+      return order[a] - order[b];
+    })
+    .map(([severity, count = 0]) => pluralizeToken(severity, count))
+    .join(', ');
+}
```

Comment:
We have a `compareIssueSeverity` function in `@code-pushup/utils` that defines the sort order.

However, it would be best to extract this whole function, since we have this logic in other plugins already, e.g.:

https://github.com/code-pushup/cli/blob/5ea6f3214557bc29c3e1c7d3b50263740b9c642b/packages/plugin-eslint/src/lib/runner/transform.ts#L52-L59